lightningd/pay.c

```diff
@@ -1498,6 +1528,70 @@ static const struct json_command listsendpays_command = {
 };
 AUTODATA(json_command, &listsendpays_command);
 
+
+static struct command_result *json_delpay(struct command *cmd,
+						const char *buffer,
+						const jsmntok_t *obj UNNEEDED,
+						const jsmntok_t *params)
+{
+	struct json_stream *response;
+	const struct wallet_payment **payments;
+	const char *status_str;
+	enum wallet_payment_status status;
+	struct sha256 *payment_hash;
+
+	if (!param(cmd, buffer, params,
+		p_req("payment_hash", param_sha256, &payment_hash),
+		p_opt("status", param_string, &status_str),
+		NULL))
+		return command_param_failed();
+
+	if (!status_str)
+		status_str = "complete";
+
+	if (!string_to_payment_status(status_str, &status))
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS, "Unrecognized status: %s", status_str);
+
+	if (status == PAYMENT_PENDING)
+		return command_fail(cmd, JSONRPC2_INVALID_PARAMS, "Invalid status: %s",
+							payment_status_to_string(status));
```

Comment:
Probably it's better to check if we have one of the valid statuses instead of not having the third. If we ever add a new status this is would not cover that fourth state. If you use a `switch` statement then the compiler will make sure we cover all cases exhaustively.

Reply:
Thanks for this suggestion, I made the change from if to switch!